packages/build/src/plugins/pinned_version.js

```diff
@@ -19,6 +20,11 @@ const pinPlugins = async function ({
   testOpts,
   sendStatus,
 }) {
+  // @todo remove this after the API bug with `updateSite` is fixed
+  if (mode !== 'require' && siteId !== 'test') {
```

Comment:
This will only evaluate to `true` when `@netlify/build` has been run programmatically (not in production nor CLI) and the `siteId` is `test`. I.e. only in automated tests.